Sources/RedisKit/RedisClient+Codable.swift

```diff
@@ -0,0 +1,40 @@
+import RediStack
+import AsyncKit
+import Foundation
+
+extension RedisClient {
+    /// Gets key as a `RedisDataConvertible` type.
+    public func get<D>(_ key: String, as type: D.Type) -> EventLoopFuture<D?> where D: RESPValueConvertible {
+        return get(key).map { value in
+            guard let value = value else {
+                return nil
+            }
+            if value.isEmpty {
+                return nil
+            } else {
+                let value = RESPValue(value)
+                return D(fromRESP: value)
+            }
+        }
+    }
+
+    // MARK: JSON
+
+    /// Gets key as a decodable type.
+    public func jsonGet<D>(_ key: String, as type: D.Type) -> EventLoopFuture<D?> where D: Decodable {
```

Comment:
Just a thought: I wonder if naming this: `get(_:asJSON:)` might be better. `jsonGet` makes it seem like that's what the actual command is, not that it's just the value being decoded / encoded as json.

Reply:
@tanner0101 so you want the function to be declared as follows?
`public func get<D>(_ key: String, asJSON type: D.Type)`

Reply:
also, what do you think of the filename?